config/routes.rb

```diff
@@ -1,8 +1,10 @@
 Rails.application.routes.draw do
   resources :contacts, only: [:index, :show, :edit, :update] do
-    resources :notes, only: [:create]
+    resources :tasks, only: [:create]
   end
-  resources :tasks, only: [:index]
+  get '/contacts/:id/tasks', to: 'contacts#show_tasks'
```

Comment:
I think there is a better way to do this, but I don't have the answer at the moment, so it's good.